.github/scripts/action.yaml

```diff
@@ -37,11 +37,34 @@ inputs:
 runs:
   using: "composite"
   steps:
+  - uses: actions/setup-python@v5
+    with:
+      python-version: 3.12
+  - name: Copy python script
+    shell: bash
+    run: |
+      set -x
+      # repository root
+      cd ${{ github.action_path }}/../../
+      rsync -rv \
+        --exclude=tests \
+        hermetic_build "${GITHUB_WORKSPACE}"
   - name: Copy shell script
     shell: bash
     run: |
       cd ${{ github.action_path }}
-      cp hermetic_library_generation.sh $GITHUB_WORKSPACE
+      cp hermetic_library_generation.sh "${GITHUB_WORKSPACE}"
+  - name: Install python packages
+    shell: bash
+    run: |
+      cd "${GITHUB_WORKSPACE}"
+      pip install --upgrade pip
```

Comment:
Do we have to upgrade `pip`? Can this be done in previous steps like `setup-python`?

Reply:
According to this [comment](https://github.com/actions/setup-python/issues/791#issuecomment-1909393984), pip is not updated in this action.

As for pip, we don't need it to be updated. I'll remove this line.

Reply:
Done.